src/codewhisperer/util/closingBracketUtil.ts

```diff
@@ -12,71 +12,74 @@ interface bracketMapType {
     [k: string]: string
 }
 
-const bracketMap: bracketMapType = {
+const quotes = ["'", '"', '`']
+const parenthesis = ['(', '[', '{', ')', ']', '}']
+
+const closeToOpenParen: bracketMapType = {
     ')': '(',
     ']': '[',
     '}': '{',
+    '>': '<',
 }
 
-export const calculateBracketsLevel = (code: string, isRightContext: boolean = false) => {
-    const bracketCounts: Map<string, number> = new Map([
-        ['(', 0],
-        ['[', 0],
-        ['{', 0],
-    ])
-    const bracketsIndexLevel = []
-
-    for (let i = 0; i < code.length; i++) {
-        const char = code[i]
-        if (bracketCounts.has(char)) {
-            const count = bracketCounts.get(char) || 0
-            const newCount = count + 1
-            bracketCounts.set(char, newCount)
-            bracketsIndexLevel.push({
-                char,
-                count: newCount,
-                idx: i,
-            })
-        } else if (char in bracketMap) {
-            const correspondingBracket = bracketMap[char as keyof bracketMapType]
-            const count = bracketCounts.get(correspondingBracket) || 0
-            const newCount = count === 0 ? 0 : count - 1
-            bracketCounts.set(bracketMap[char], newCount)
-            bracketsIndexLevel.push({
-                char: correspondingBracket,
-                count: newCount,
-                idx: i,
-            })
-        } else if (isRightContext && !(char in bracketMap) && !bracketCounts.has(char) && !/\s/.test(char)) {
-            // we can stop processing right context when we encounter a char that is not a bracket nor white space
-            break
-        }
-    }
-    return bracketsIndexLevel
+const openToCloseParen: bracketMapType = {
+    '(': ')',
+    '[': ']',
+    '{': '}',
+    '<': '>',
```

Comment:
`parenthesis` does not have `<` and `openToCloseParen` has `<`, is this intended?

Reply:
oh no thanks for catching this

Reply:
will attach some tests gif in the PR

Reply:
this should be finalized now,  i think it already solve most of scenarios